src/emulator/cpu.c

```diff
@@ -937,17 +937,26 @@ static bool cpuMakeDevice(Cpu* pCPU, s32* piDevice, void* pObject, u32 nOffset,
                 pDevice->nAddressPhysical1 = 0xFFFFFFFF;
                 pDevice->nAddressVirtual1 = 0xFFFFFFFF;
 
+#if VERSION == OOT_U
+                for (j = 0; j < 0x100000; j++) {
+                    pCPU->aiDevice[j] = iDevice;
+                }
+#else
                 for (j = 0; j < 0x10000; j++) {
                     pCPU->aiDevice[j] = iDevice;
                 }
+#endif
             } else {
                 pDevice->nAddressVirtual0 = nOffset;
                 pDevice->nAddressVirtual1 = nOffset + nSize - 1;
                 pDevice->nAddressPhysical0 = nAddress;
                 pDevice->nAddressPhysical1 = nAddress + nSize - 1;
-
                 for (j = nSize; j > 0; nOffset += 0x10000, j -= 0x10000) {
+#if VERSION == OOT_U
+                    pCPU->aiDevice[nOffset >> 12] = iDevice;
+#else
                     pCPU->aiDevice[nOffset >> 16] = iDevice;
+#endif
```

Comment:
We should probably give a name to 12/16 since it will have to be used everywhere aiDevice is used. (also the CPU_DEVICE should probably be updated too.) Maybe

```
#if VERSION == OOT_U
#define DEVICE_INDEX_BITS 16
#else
#define DEVICE_INDEX_BITS 20
#endif

#define DEVICE_OFFSET_BITS (32 - DEVICE_INDEX_BITS)
```
and then we can use `nOffset >> DEVICE_OFFSET_BITS` everywhere and aiDevice can have size `1 << DEVICE_INDEX_BITS`.

Reply:
actually `DEVICE_ADDRESS_INDEX_BITS` and `DEVICE_ADDRESS_OFFSET_BITS` are probably more accurate but it's getting long. Someday we should try to explain them too but that can be TBD